x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/terms/values_range_input.tsx

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useState } from 'react';
+import { useDebounce } from 'react-use';
+import { i18n } from '@kbn/i18n';
+import { EuiRange } from '@elastic/eui';
+
+type PropType<C> = C extends React.ComponentType<infer P> ? P : unknown;
+// Add ticks to EuiRange component props
+const FixedEuiRange = (EuiRange as unknown) as React.ComponentType<
+  PropType<typeof EuiRange> & {
+    ticks?: Array<{
+      label: string;
+      value: number;
+    }>;
+  }
+>;
+
+export const ValuesRangeInput = ({
+  value,
+  onChange,
+}: {
+  value: number;
+  onChange: (value: number) => void;
+}) => {
+  const MIN_NUMBER_OF_VALUES = 1;
+  const MAX_NUMBER_OF_VALUES = 100;
+
+  const [inputValue, setInputValue] = useState(String(value));
+  useDebounce(
+    () => {
+      if (inputValue === '') {
+        return;
+      }
+      const inputNumber = Number(inputValue);
+      onChange(Math.min(MAX_NUMBER_OF_VALUES, Math.max(inputNumber, MIN_NUMBER_OF_VALUES)));
+    },
+    256,
+    [inputValue]
+  );
+
+  return (
+    <FixedEuiRange
+      min={MIN_NUMBER_OF_VALUES}
+      max={MAX_NUMBER_OF_VALUES}
+      step={1}
+      value={inputValue}
+      showInput
+      showLabels
+      compressed
+      onChange={({ currentTarget }) => setInputValue(currentTarget.value)}
+      aria-label={i18n.translate('xpack.lens.indexPattern.terms.size', {
+        defaultMessage: 'Number of values',
+      })}
+    />
```

Comment:
I found it confusing that I could be in an invalid state as long as I had focus, and the red highlighting would only appear once I blur the input:

<img width="418" alt="Screen Shot 2020-09-29 at 3 05 41 PM" src="https://user-images.githubusercontent.com/666475/94604975-6cc98d00-0266-11eb-91e0-3c89b8fb7a80.png">

Reply:
Yeah, I noticed that too, but this is the internal behaviour of EuiRange - even adding `isInvalid` prop with some conditions doesn't help - the condition 'is fired' on blur and not on change, so it won't help. I'll speak to eui team to see if they can do something about. For now, do you think it's a blocker to merge this one?

Reply:
Opening an issue for EUI is worth it, but it shouldn't block the PR IMHO